2. Web API now calls Microsoft Graph/TodoListService/Controllers/TodoListController.cs

```diff
@@ -56,17 +35,16 @@ public TodoListController(ITokenAcquisition tokenAcquisition)
         static readonly ConcurrentBag<TodoItem> TodoStore = new ConcurrentBag<TodoItem>();
 
         /// <summary>
-        /// The Web API will only accept tokens 1) for users, 2) having the user_impersonation scope for
-        /// this API (if you created the app using the App creation script) or 'access_as_user' if you created
-        /// it following the README.md.
+        /// The Web API will only accept tokens 1) for users, and 
+        /// 2) having the user_impersonation scope for this API
         /// </summary>
-        static string[] scopeRequiredByAPI = new string[] { "user_impersonation", "access_as_user" };
```

Comment:
Please check in the README.md. When registrating the Web API in Azure AD with AppCreationScripts, the scope will be user_impersonation (no choice it's a v1.0 registration for the moment as AppModel/PowerShell does not yet support v2.0), If the Web API is registrerd with the UX the natural scope is access_as_user.

We could force user_impersonation,b but would want to change the README, then?

The simplest migt be the leave both scopes?

Reply:
Yes...i did check the ReadMe.md, but when using AppCreationScripts, as you mention, you use one scope but not the other. It just gets confusing having both scopes there, as a developer/customer, it's not clear why/how they are used. Which is why i removed access_as_user. I would suggest just updating the ReadMe.md. I can do that if you want.

Also, developers don't have to use the scope "user_impersonation", the actual name of the scope could be anything.

---
In reply to: [318905693](https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2/pull/93#discussion_r318905693) [](ancestors = 318905693)

Reply:
Correct. Please avoid using user_impersonation. It is an anti-pattern as we used to issue it by default for 1P apps. 
WebAPI should register its own scope.

Reply:
Thanks for the heads-up, @yordan-msft. We are changing the app creation script generator to add access_as_user.